### ISSUE TITLE
Suggestion: overwrite no-overwrite feature

### ISSUE THREAD

**Issue by caligo-erik** [fmvilas/openapi3-generator#16]
First of all: great tool, I am using it for a new project of mine and these "dynamic templates" are really helpful.

To make this tool even greater, I'd suggest the following features:
1. don't-overwrite flag
2. delete all folders called 'auto'
#### This is what I want to achieve with the suggested features:
- files containing "dumb code", e.g. ```routes/auto/$$path$$.js``` will be overwritten each time
- files destined for business logic e.g. ```services/$$path$$.js``` will be written, but not overwritten.
#### Advantages:
- by deleting the 'auto' folders each time, all unused/old files will be deleted.
- business-logic-code-files will be preserved. No more accidental losses. If you want to to re-generate a specific file, e.g. to track changes in the specs, simply move the file to another folder, re-run the code generator, and start merging the changes. For more drastic measures, just remove the _don't overwrite flag_ and all files will be overwritten as usual.

**Comment by fmvilas** [comment]
I think this is a great idea, especially the "don't overwrite flag". The only thing I don't like about the first idea is the "auto" directory because that conditions the structure of the template when it shouldn't. What about passing another flag with a list of directories you want to "empty" each time? You could have something like this:

```
og -o ./my-docs --rm the-dumb-code --rm the-other-dumb-code-directory --no-overwrite services/* openapi.yaml express 
```

What do you think?

**Comment by caligo-erik** [comment]
> I think this is a great idea, especially the "don't overwrite flag". The only thing I don't like about the first idea is the "auto" directory because that conditions the structure of the template when it shouldn't. What about passing another flag with a list of directories you want to "empty" each time? You could have something like this:
> 
> ```
> og -o ./my-docs --rm the-dumb-code --rm the-other-dumb-code-directory --no-overwrite services/* openapi.yaml express 
> ```
> What do you think?

I'd prefer the recursive delete-folder feature, since it requires less maintenance effort. The delete parameter is specified only once and can be copied to other projects without any problems. This makes it easy to impose as a _standard_ for your dev team.

I see it less a _conditioning of the structure_ and more of a _providing useful information_ such as: **"you are in editing code that is likely to be overwritten. Nothing important happens here."**
Example: 
- api
  - data
    - mocks
      - auto
        - path1.js
        - path2.js
    - database
      - auto
        - path1.js
        - path2.js
      - databaseHelper.js
    - dataProvider.js
  - routes
    - auto
      - path1.js
      - path2.js
  - services
  	- path1.js
    - path2.js
  - validation
    - auto
      - path1.js
      - path2.js

**Comment by fmvilas** [comment]
Fixed in #17. Thanks!
